duqtools/operations.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+import logging
+from collections import deque
+from inspect import signature
+from typing import Callable
+
+import click
+from pydantic import Field
+
+from .schema import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+class Operation(BaseModel):
+    """Operation, simple class which has a callable action."""
+
+    description: str = Field(
+        description='description of the operation to be done')
+    action: Callable = Field(
+        description='a function which can be executed when we '
+        'decide to apply this operation')
+    args: tuple = Field((),
+                        description='positional arguments that have to be '
+                        'passed to the action')
+    kwargs: dict = Field({},
+                         description='keyword arguments that will be '
+                         'passed to the action')
+
+    def __call__(self) -> Operation:
+        logger.info(f'-: {self.description}')
```

Comment:
Why not just log the description here?

```suggestion
        logger.info(self.description)
```